pkg/apis/certmanager/v1alpha1/types_issuer.go

```diff
@@ -90,6 +90,8 @@ type VaultIssuer struct {
 	Server string `json:"server"`
 	// Vault URL path to the certificate role
 	Path string `json:"path"`
+	// CA bundle to validate Vault server certificate
+	CABundle string `json:"caBundle"`
```

Comment:
This should be a []byte, inline with other examples of caBundle in Kubernetes. Users then specify it as a base64 encoded value in their YAML :smile: (we should also update the docs to mention this too)

Reply:
Can you also add some resource validation that attempts to decode this bundle and ensures it is valid?

Reply:
Also, can you update the comment (and other doc above) to clarify the behaviour if:

- A user specifies a https:// URL, but no caBundle
- A user specifies a https:// URL, *with* a caBundle (i.e. call out that system roots are not used at all)
- A user specifies a http:// URL, *with* a caBundle